homeassistant/components/tplink/__init__.py

```diff
@@ -136,33 +147,66 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             ", ".join(d.host for d in switches),
         )
 
+    async def async_retry_devices(self) -> None:
+        """Retry unavailable devices."""
+        unavailable_devices: list[SmartDevice] = hass_data[UNAVAILABLE_DEVICES]
+        _LOGGER.debug(
+            "retry during setup unavailable devices: %s",
+            [d.host for d in unavailable_devices],
+        )
+
+        for device in unavailable_devices:
+            try:
+                device.get_sysinfo()
+            except SmartDeviceException:
+                continue
+            _LOGGER.debug(
+                "at least one device is available again, so reload integration"
+            )
+            await hass.config_entries.async_reload(entry.entry_id)
```

Comment:
No need to reload multiple times if 2 devices come online.

```suggestion
            await hass.config_entries.async_reload(entry.entry_id)
            break
```